### ISSUE TITLE
[a11y]: Loading-  Notifying Screen Reader Users when "loading" is completed 

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16415]
### Package

@carbon/react

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

When the `active` prop is `false`, Visually loading symbol disappears but SR still announcing as "Loading".

https://github.com/carbon-design-system/carbon/assets/63502271/a7bd2d33-d1dc-4a17-8b9e-f1ae46dfbfbe

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-loading--playground

### Recommend Fix
below are two approaches  

- The loader should have aria-busy set to false when it is set as false
- For screen reader users, it's helpful to provide an "Loading complete" aria-status message when the spinning wheel disappears. This notifies them when the page is ready, saving them from continuously checking its status.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
The second approach could be something like this

https://github.com/carbon-design-system/carbon/blob/f0d89db496ae5f16a18a43b642b698c2173ac608/packages/react/src/components/TextArea/TextArea.tsx#L461-L463
